ChatSecure/Classes/View Controllers/OTRMessagesViewController.m

```diff
@@ -1632,13 +1647,15 @@ - (NSString *)senderDisplayName
     if ([message isKindOfClass:[OTRXMPPRoomMessage class]]) {
         OTRXMPPRoomMessage *roomMessage = (OTRXMPPRoomMessage *)message;
         __block OTRXMPPRoomOccupant *roomOccupant = nil;
-        [self.readOnlyDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction * _Nonnull transaction) {
-            [transaction enumerateRoomOccupantsWithJid:roomMessage.senderJID block:^(OTRXMPPRoomOccupant * _Nonnull occupant, BOOL * _Null_unspecified stop) {
-                roomOccupant = occupant;
-                *stop = YES;
-            }];
+        [self.readOnlyDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *_Nonnull transaction) {
+            roomOccupant = [self occupantWithTransaction:transaction forFullJid:roomMessage.senderJID inRoom:roomMessage.roomUniqueId];
         }];
-        UIImage *avatarImage = [roomOccupant avatarImage];
+        UIImage *avatarImage = nil;
+        if (roomOccupant) {
+            avatarImage = [roomOccupant avatarImage];
```

Comment:
Above you can fetch the realBuddy using the func, and then just use the avatarImage of the buddy if it's available.